raylib/extraction/raytrees.cpp

```diff
@@ -1051,11 +1057,14 @@ void Trees::removeOutOfBoundSections(const Cloud &cloud, Eigen::Vector3d &min_bo
   const double width = params_->grid_width;
   cloud.calcBounds(&min_bound, &max_bound);
   const Eigen::Vector3d mid = (min_bound + max_bound)/2.0 + offset;
-  const Eigen::Vector2d inds(std::round(mid[0] / width), std::round(mid[1] / width));
-  min_bound[0] = width * (inds[0] - 0.5) - offset[0];
-  min_bound[1] = width * (inds[1] - 0.5) - offset[1];
-  max_bound[0] = width * (inds[0] + 0.5) - offset[0];
-  max_bound[1] = width * (inds[1] + 0.5) - offset[1];
+  Eigen::Vector2d mid_local(mid[0], mid[1]);
+  mid_local -= params_->grid_origin;
+  const Eigen::Vector2d inds(std::floor(mid_local[0] / width), std::floor(mid_local[1] / width));
+  min_bound[0] = width * (inds[0]) + params_->grid_origin[0] - offset[0];

```

Comment:
If I am understanding your implementation correctly, you don't need to add the params_->grid_origin[0], for subsequent calls as well, otherwise, you remove the grid origin shift?

Reply:
Basically it is: grid_origin + snap_to_grid(mid - grid_origin)
which means snapping to an offset grid.
I've tested it on simple cases and seems to work.